types/react/index.d.ts

```diff
@@ -1737,9 +1741,17 @@ declare namespace React {
      * A {@link Dispatch} function can sometimes be called without any arguments.
      */
     type DispatchWithoutAction = () => void;
-    // Unlike redux, the actions _can_ be anything
+    /**
+     * A function to manage the state updates {@link useReducer} hook
+     * Unlike redux, the actions _can_ be anything
+     * @template S - The type of the state
+     * @template A - The type of the action
+     */
     type Reducer<S, A> = (prevState: S, action: A) => S;
-    // If useReducer accepts a reducer without action, dispatch may be called without any parameters.
+    /**
+     * A {@link Reducer} function can sometimes be defined without any actions
```

Comment:
It should explain when you want to do that. 

This might be an instance of a internal utility for type inference and not something you'd use in product code. If you don't use it in product code, we don't need to add a description.